package/lib/src/utils.dart

```diff
@@ -34,93 +35,120 @@ abstract class Utils {
-  static BeamState createBeamState(
-    Uri uri, {
-    BeamLocation? beamLocation,
-    Map<String, dynamic> data = const <String, dynamic>{},
-  }) {
-    if (beamLocation != null) {
-      for (var pathBlueprint in beamLocation.pathBlueprints) {
+      if (pathBlueprint is String) {
         if (pathBlueprint == uri.path || pathBlueprint == '/*') {
-          BeamState(
-            pathBlueprintSegments: uri.pathSegments,
-            queryParameters: uri.queryParameters,
-            data: data,
-          );
+          return true;
         }
         final uriPathSegments = List.from(uri.pathSegments);
         if (uriPathSegments.length > 1 && uriPathSegments.last == '') {
           uriPathSegments.removeLast();
         }
         final beamLocationPathBlueprintSegments =
             Uri.parse(pathBlueprint).pathSegments;
-        var pathSegments = <String>[];
-        var pathParameters = <String, String>{};
         if (uriPathSegments.length > beamLocationPathBlueprintSegments.length &&
             !beamLocationPathBlueprintSegments.contains('*')) {
           continue;
         }
         var checksPassed = true;
         for (int i = 0; i < uriPathSegments.length; i++) {
           if (beamLocationPathBlueprintSegments[i] == '*') {
-            pathSegments = List<String>.from(uriPathSegments);
             checksPassed = true;
             break;
           }
           if (uriPathSegments[i] != beamLocationPathBlueprintSegments[i] &&
               beamLocationPathBlueprintSegments[i][0] != ':') {
             checksPassed = false;
             break;
-          } else if (beamLocationPathBlueprintSegments[i][0] == ':') {
-            pathParameters[beamLocationPathBlueprintSegments[i].substring(1)] =
-                uriPathSegments[i];
-            pathSegments.add(beamLocationPathBlueprintSegments[i]);
-          } else {
-            pathSegments.add(uriPathSegments[i]);
           }
         }
         if (checksPassed) {
-          return BeamState(
-            pathBlueprintSegments: pathSegments,
-            pathParameters: pathParameters,
-            queryParameters: uri.queryParameters,
-            data: data,
-          );
+          return true;
+        }
+      } else {
+        final regexp = tryCastToRegExp(pathBlueprint);
+        return regexp.hasMatch(uri.toString());
+      }
+    }
+    return false;
+  }
+
+  /// Creates a state for [BeamLocation] based on incoming [uri].
+  ///
+  /// Used in [BeamState.copyForLocation].
+  static BeamState createBeamState(
+    Uri uri, {
+    BeamLocation? beamLocation,
+    Map<String, dynamic> data = const <String, dynamic>{},
+  }) {
+    if (beamLocation != null) {
+      // TODO: abstract this and reuse in canBeamLocationHandleUri
+      for (var pathBlueprint in beamLocation.pathBlueprints) {
+        if (pathBlueprint is String) {
+          if (pathBlueprint == uri.path || pathBlueprint == '/*') {
+            BeamState(
```

Comment:
I'm debugging an issue with one of my routes and came across this line of code (not sure if related or not).

But what is the purpose of constructing a `BeamState` and nothing uses the reference?